scripts/support/zephyr_flash_debug.py

```diff
@@ -38,105 +40,236 @@ def get_env_bool_or(env_var, default_value):
         return default_value
 
 
+def get_env_strip_or(env_var, to_strip, default_value):
+    value = os.environ.get(env_var, None)
+    if value is not None:
+        return value.strip(to_strip)
+    else:
+        return default_value
+
+
+def quote_sh_list(cmd):
+    '''Transform a command from list into shell string form.'''
+    fmt = ' '.join('{}' for _ in cmd)
+    args = [shlex.quote(s) for s in cmd]
+    return fmt.format(*args)
+
+
 def check_call(cmd, debug):
     if debug:
-        print(' '.join(cmd))
+        print(quote_sh_list(cmd))
     subprocess.check_call(cmd)
 
 
 def check_output(cmd, debug):
     if debug:
-        print(' '.join(cmd))
+        print(quote_sh_list(cmd))
     return subprocess.check_output(cmd)
 
 
-class ZephyrBinaryFlasher(abc.ABC):
-    '''Abstract superclass for flasher objects.'''
+def popen_ignore_int(cmd, debug):
+    '''Spawn a child command, ensuring it ignores SIGINT.
+
+    The returned subprocess.Popen object must be manually terminated.'''
+    cflags = 0
+    preexec = None
+    system = platform.system()
+
+    if system == 'Windows':
+        cflags |= subprocess.CREATE_NEW_PROCESS_GROUP
+    elif system in {'Linux', 'Darwin'}:
```

Comment:
What's the value of `os.setsid` on Windows? Might be the case that, if it's available, use it.